GW2EIEvtcParser/EncounterLogic/Raids/W8/GreerTheBlightbringer.cs

```diff
@@ -137,4 +234,238 @@ internal override void ComputePlayerCombatReplayActors(PlayerActor player, Parse
+            }
+        }
+
+        // Cage of Decay - Moving roots walls
+        if (log.CombatData.TryGetEffectEventsBySrcWithGUID(target.AgentItem, EffectGUIDs.GreerCageOfDecayMovingWalls, out var cageOfDecayWalls))
+        {
+            foreach (EffectEvent effect in cageOfDecayWalls)
+            {
+                (long start, long end) lifespan = effect.ComputeLifespan(log, 2000);
+                var wall = (RectangleDecoration)new RectangleDecoration(100, 50, lifespan, Colors.GreenishYellow, 0.3, new PositionConnector(effect.Position)).UsingRotationConnector(new AngleConnector(effect.Rotation.Z));
+                replay.Decorations.AddWithBorder(wall, Colors.Purple, 0.2);
+            }
+        }
+
+        // Cage of Decay - Roots walls around the circle of damage
+        if (log.CombatData.TryGetEffectEventsBySrcWithGUID(target.AgentItem, EffectGUIDs.GreerCageOfDecayCircleWalls, out var cageOfDecayCircleWalls))
+        {
+            foreach (EffectEvent effect in cageOfDecayCircleWalls)
+            {
+                (long start, long end) lifespan = effect.ComputeLifespan(log, 23000);
+                var wall = (RectangleDecoration)new RectangleDecoration(200, 100, lifespan, Colors.GreenishYellow, 0.3, new PositionConnector(effect.Position)).UsingRotationConnector(new AngleConnector(effect.Rotation.Z));
+                replay.Decorations.AddWithBorder(wall, Colors.Purple, 0.2);
+            }
+        }
+    }
+
+    private static void AddRainOfSpores(NPC target, ParsedEvtcLog log, CombatReplay replay)
+    {
+        // Rain of Spores - Indicator
+        if (log.CombatData.TryGetEffectEventsBySrcWithGUID(target.AgentItem, EffectGUIDs.GreerRainOfSporesIndicator, out var rainOfSpores))
+        {
+            foreach (EffectEvent effect in rainOfSpores)
+            {
+                (long start, long end) lifespan = effect.ComputeLifespan(log, effect.Duration);
+                replay.Decorations.Add(new CircleDecoration(200, lifespan, Colors.LightOrange, 0.2, new PositionConnector(effect.Position)));
+            }
+        }
+    }
+
+    private static void AddStompTheGrowth(NPC target, ParsedEvtcLog log, CombatReplay replay, uint radius)
+    {
+        // Stomp the Growth - Circle indicator
+        if (log.CombatData.TryGetEffectEventsBySrcWithGUID(target.AgentItem, EffectGUIDs.GreerStompTheGrowth, out var stompTheGrowth))
+        {
+            foreach (EffectEvent effect in stompTheGrowth)
+            {
+                (long start, long end) lifespan = effect.ComputeLifespan(log, effect.Duration);
+                var circle = new CircleDecoration(radius, lifespan, Colors.LightOrange, 0.2, new PositionConnector(effect.Position));
+                replay.Decorations.AddWithGrowing(circle, effect.Time + effect.Duration);
+            }
+        }
+
+        // Gree doesn't create shockwaves
+        if (target.IsSpecies(ArcDPSEnums.TargetID.Greer))
```

Comment:
This should be put outside of this API, the API below does not make use of target nor radius, which makes this section of AddStompTheGrowth non-generic

Reply:
Could be actually put in Environment decorations.